packages/compiler/src/core/types.ts

```diff
@@ -2889,13 +2889,25 @@ export interface RelatedSourceLocation {
 
 export interface LogSink {
   log(log: ProcessedLog): void;
+  trackAction<T>(
```

Comment:
I think this is nearly there, we might want to make this value optional and wrap around in case its not set just because otherwise its maybe a bit too much of a breaking change. It also doesn't feel necessary to have to define it.